raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -877,10 +866,11 @@ private FetchQuorumRecordsResponseData buildEmptyFetchQuorumRecordsResponse(
             return completedFuture(response);
         }
 
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        purgatory.await(future, request.maxWaitTimeMs());
+        CompletableFuture<Long> future = new CompletableFuture<>();
+        // all fetch requests can be satisfied even with zero bytes
+        fetchPurgatory.await(future, 0, request.maxWaitTimeMs());
```

Comment:
Perhaps we may as well change the fetch purgatory so that it is based on the fetch offset?